modules/default/calendar/calendar.js

```diff
@@ -559,7 +559,7 @@ Module.register("calendar", {
 			let maxPastDaysCompare = now - this.maximumPastDaysForUrl(calendarUrl) * ONE_DAY;
 			for (const e in calendar) {
 				const event = JSON.parse(JSON.stringify(calendar[e])); // clone object
-
+				event.url = calendarUrl;
```

Comment:
this is because otherwise the url isn't set yet for the comparison in listContainsEvent() a few lines below. 

But a better solution would be to compare events based on their UID. But for this I'd need more code (and javascript :D) knowledge